.github/workflows/tests.yml

```diff
@@ -9,52 +9,106 @@ on:
+          run-post: false
 
-      - name: Install dependencies constructor-manager-cli
+      - name: Install dependencies on base environment
         run: |
-          python -m pip install --upgrade pip
-          python -m pip install setuptools tox tox-gh-actions
-          cd constructor-manager-cli
-          pip install -e .
-          pip list
+          conda install -n base conda-lock mamba -c conda-forge --quiet
+          conda list
 
-      - name: Test constructor-manager-cli
+      - name: create constructor-manager-environment
         run: |
-          cd constructor-manager-cli
-          python -m tox
-        env:
-          PLATFORM: ${{ matrix.platform }}
+          conda install -n constructor-manager conda packaging requests pyyaml -c conda-forge --quiet
+          # List installed packages
+          conda list
 
-      - name: Install dependencies constructor-manager
+      - name: Install constructor-manager
         run: |
-          cd constructor-manager-cli
-          pip install -e .
-          pip list
-        env:
-          PLATFORM: ${{ matrix.platform }}
+          # Install constructor manager
+          git clone https://github.com/goanpeca/packaging.git packaging_clone
+          cd packaging_clone
+          git checkout constructor-updater
+          cd constructor-manager
+          pip install -e . --no-deps
 
-      - name: Test constructor-manager
+          # Install test deps
+          conda install -n constructor-manager pytest pytest-cov pytest-qt -c conda-forge --quiet
+
+          # List installed packages
+          conda list
+
+      - name: Install constructor-manager-api
         run: |
-          cd constructor-manager
-          python -m tox
-        env:
-          PLATFORM: ${{ matrix.platform }}
+          # Install dependencies
+          conda install -n constructor-manager qtpy pyqt -c conda-forge --quiet
+
+          # Install constructor manager
+          cd packaging_clone
+          git checkout constructor-cli
+          cd constructor-manager-api
+          pip install -e . --no-deps
+
+          # List installed packages
+          conda list
+
+      - name: Install constructor-manager-ui
+        run: |
+          # Install constructor manager ui
+          cd constructor-manager-ui
+          pip install -e . --no-deps
+
+          # List installed packages
+          conda list
+
+      - name: List installed packages
+        run: |
+          conda list
+
+      - name: Test constructor-manager-ui (linux)
+        if: contains(matrix.platform, 'ubuntu')
+        run: |
+          # Run Tests
+          cd constructor-manager-ui/src
+          xvfb-run pytest constructor_manager_ui --cov=constructor_manager_ui
+
+      - name: Test constructor-manager-ui (other)
+        if: "!contains(matrix.platform, 'ubuntu')"
+        run: |
+          # Run Tests
+          cd constructor-manager-ui/src
+          pytest constructor_manager_ui --cov=constructor_manager_ui
```

Comment:
I guess the UI part will not use `tox` for the tests on the CI? Should the `tox.ini` file be removed?

Reply:
Yes, since all the workflow for these packages revolves around constructor using a base environment and a constructor-manager environment, tox does not help a lot. Will remove!